workers/unity/Packages/com.improbable.gdk.deploymentlauncher/.DeploymentLauncher/Commands/Create.cs

```diff
@@ -0,0 +1,206 @@
+            where TOptions : Options.Create
+        {
+            var snapshotServiceClient = SnapshotServiceClient.Create();
+            var deploymentServiceClient = DeploymentServiceClient.Create();
+
+            try
+            {
+                var deployment = new Deployment
+                {
+                    AssemblyId = options.AssemblyName,
+                    LaunchConfig = new LaunchConfig
+                    {
+                        ConfigJson = getLaunchConfigJson(options)
+                    },
+                    Name = options.DeploymentName,
+                    ProjectName = options.ProjectName,
+                    RegionCode = options.Region.ToString()
+                };
+
+                if (options.SnapshotPath != null)
+                {
+                    var snapshotId = UploadSnapshot(snapshotServiceClient, options.SnapshotPath, options.ProjectName,
+                        options.DeploymentName);
+
+                    if (string.IsNullOrEmpty(snapshotId))
+                    {
+                        return Program.ErrorExitCode;
+                    }
+
+                    deployment.StartingSnapshotId = snapshotId;
+                }
+
+                if (options.Tags != null)
+                {
+                    foreach (var tag in options.Tags)
+                    {
+                        deployment.Tag.Add(tag);
+                    }
+                }
+
+                var deploymentOp = deploymentServiceClient.CreateDeployment(new CreateDeploymentRequest
+                {
+                    Deployment = deployment
+                }).PollUntilCompleted();
+
+                if (deploymentOp.Result.Status != Deployment.Types.Status.Running)
+                {
+                    Ipc.WriteError(Ipc.ErrorCode.Unknown, "Deployment failed to start for an unknown reason.");
+                    return Program.ErrorExitCode;
+                }
+            }
+            catch (Grpc.Core.RpcException e)
+            {
+                if (e.Status.StatusCode == Grpc.Core.StatusCode.NotFound)
+                {
+                    Ipc.WriteError(Ipc.ErrorCode.NotFound, e.Status.Detail);
+                    return Program.ErrorExitCode;
+                }
+
+                throw;
+            }
+
+            return Program.SuccessExitCode;
+        }
+
+        private static string ModifySimulatedPlayerLaunchJson(Options.CreateSimulated options)
+        {
+            var playerAuthServiceClient = PlayerAuthServiceClient.Create();
+
+            // Create development authentication token used by the simulated players.
+            var dat = playerAuthServiceClient.CreateDevelopmentAuthenticationToken(
+                new CreateDevelopmentAuthenticationTokenRequest
+                {
+                    Description = "DAT for sim worker deployment.",
+                    Lifetime = Duration.FromTimeSpan(TimeSpan.FromDays(7)),
```

Comment:
this should be configurable

Reply:
Agreed - will open a ticket to add this as an option

Reply:
https://improbableio.atlassian.net/browse/UTY-1964